lib/service/database.js

```diff
@@ -284,6 +288,12 @@ DatabaseService.prototype.queueAnalysisOperations = function(analysis, callback)
                 asyncQueries.push({
                     query: updateNodeStatusAtAnalysisCatalogQuery(nodeIds, Node.STATUS.RUNNING)
                 });
+                if (sqlWrappedNode.preCheckNodeQuery() !== undefined) {
+                    asyncQueries.push({
+                        query: readOnlyTransactionQuery([sqlWrappedNode.preCheckNodeQuery()]),
```

Comment:
From my PoV we should. A check should not perform a write operation, so enforce it could be a life saver.